src/main/java/com/boraji/tutorial/spring/controller/ProductsController.java

```diff
@@ -0,0 +1,120 @@
+        }
+        return "redirect:/products/store";
+    }
+
+    @RequestMapping(value = "/edit", method = RequestMethod.GET)
+    public String updateProduct(@RequestParam("productID") String id,
+                                @ModelAttribute("user") User user,
+                                Model model) {
+        try {
+            if (!Objects.isNull(user.getId()) && user.getRole().equals("admin")) {
+                Optional<Product> optionalProduct = productService.getById(Long.valueOf(id));
+                if (optionalProduct.isPresent()) {
+                    Product editProduct = optionalProduct.get();
+                    model.addAttribute("id", editProduct.getId());
+                    model.addAttribute("name", editProduct.getName());
+                    model.addAttribute("description", editProduct.getDescription());
+                    model.addAttribute("price", editProduct.getPrice());
+                    return "edit_product";
+                }
+            }
+        } catch (NumberFormatException e) {
+            return "redirect:/products/store";
+
+        }
+        return "redirect:/products/store";
+    }
+
+    @RequestMapping(value = "/edit", method = RequestMethod.POST)
+    public String editProduct(@RequestParam("productID") String id,
+                              @RequestParam String name,
+                              @RequestParam String description,
+                              @RequestParam String price,
+                              @ModelAttribute("user") User user) {
+        try {
+            if (!Objects.isNull(user.getId()) &&
+                    user.getRole().equals("admin")) {
+                if (productService.getById(Long.valueOf(id)).isPresent()) {
+                    productService.updateProduct(Long.valueOf(id), name, description, Double.valueOf(price));
+                }
+            }
+        } catch (NumberFormatException e) {
+            return "redirect:/products/store";
+        }
+        return "redirect:/products/store";
+    }
+
+    @RequestMapping(value = "/add", method = RequestMethod.GET)
+    public String addProductView(@ModelAttribute("user") User user) {
+        if (!Objects.isNull(user.getId()) && user.getRole().equals("admin")) {
+            return "add_product";
+        }
+        return "redirect:/products/store";
+    }
+
+
+    @RequestMapping(value = "/add", method = RequestMethod.POST)
+    public String addProduct(@RequestParam String name,
+                             @RequestParam String description,
+                             @RequestParam String price,
+                             @ModelAttribute("user") User user,
+                             Model model) {
+        System.out.println(user.getRole());
+        try {
+            if (!Objects.isNull(user.getId()) && user.getRole().equals("admin")) {
+                productService.add(name, description, Double.valueOf(price));
```

Comment:
get from page Product model, and service accepts model, not String args

Reply:
I think its will be incorrect cause un service I validate all that fields